### PR TITLE
move ParseToken() function to cluster-bootstrap/token/util package

### PR DESCRIPTION
**Pull request by boddumanohar**
/kind cleaup

**What this PR does / why we need it**:
The `staging/src/k8s.io/cluster-bootstrap/token/util/` package a has some utility functions in it. But there is also `staging/src/k8s.io/cluster-bootstrap/utils/` which is supposed to be the home of all the utility related functions within cluster-bootstrap so moved everything to util folder.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
To reviewers, I have a question, what is the difference between `package tokens` and `package secrets` in cluster-bootstrap/util folder
  
**Does this PR introduce a user-facing change?**:
NO

```release-note
NONE
```
cc: @dixudx

/sig-cluster-lifecycle

### CODE REVIEW

File: cmd/kubeadm/app/apis/kubeadm/bootstraptokenhelpers.go [hunk 1]
```diff
@@ -26,16 +26,16 @@ import (
 	"k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	bootstrapapi "k8s.io/cluster-bootstrap/token/api"
-	bootstraputil "k8s.io/cluster-bootstrap/token/util"
+	bootstraptokenutil "k8s.io/cluster-bootstrap/util/tokens"
```
Comment (neolit123) [inline cmd/kubeadm/app/apis/kubeadm/bootstraptokenhelpers.go]: my preference would be to move this the other way around....

cluster-bootstrap/util/tokens/tokens.go has a single function:
`ParseToken()`

move that to:
`cluster-bootstrap/token/util/helpers.go`

and add a unit test for it in helpers_test.go.

**Comment by neolit123** [comment]
> To reviewers, I have a question, what is the difference between package tokens and package secrets in cluster-bootstrap/util folder

touching secrets should be a separate PR.

tokens and secretes are separate API objects.
maybe we want to have the following directory structure:
- tokens
  - utils
- secrets
  - utils

instead of a parent `/utils`...

**Comment by boddumanohar** [comment]
@neolit123 I've moved ParseToken() function to cluster-bootstrap/token/util package and added a testcase. 

Thanks,
Manohar.

**Comment by neolit123** [comment]
/kind cleanup
/priority important-soon

i'm +1 on the change as is, but it touches areas that i cannot approve.
/lgtm

**Comment by neolit123** [comment]
@boddumanohar 

> ERROR: /home/prow/go/src/github.com/kubernetes/kubernetes/staging/src/k8s.io/cluster-bootstrap/BUILD:8:1: no such package 'staging/src/k8s.io/cluster-bootstrap/token/util': BUILD file not found on package path and referenced by '//staging/src/k8s.io/cluster-bootstrap:all-srcs'
ERROR: Analysis of target '//test/instrumentation:verify_stable_metric' failed; build aborted: no such package 'staging/src/k8s.io/cluster-bootstrap/token/util': BUILD file not found on package path

https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/88665/pull-kubernetes-bazel-test/1233404528523481088

try running  ./hack/update-bazel.sh and commiting all BUILD changes.

also there are some other errors from gofmt, lint and staticcheck:
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/88665/pull-kubernetes-verify/1233404528586395651

./hack/update-gofmt.sh can fix the fmt issues.

**Comment by neolit123** [comment]
looks like you need to call:
./hack/update-vendor.sh
./hack/update-gofmt.sh

**Comment by boddumanohar** [comment]
even after running `./hack/update-gofmt.sh` & `./hack/update-vendor.sh`  I see one of the test case has failed. any idea how to fix that?

https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/88665/pull-kubernetes-verify/1233757001041842177

**Comment by BenTheElder** [comment]
/approve
for vendor/

**Comment by enj** [comment]
> maybe we want to have the following directory structure:
> 
>     * tokens
>       
>       * utils
> 
>     * secrets
>       
>       * utils
> 
> 
> instead of a parent `/utils`...

This type of structure is generally discouraged because it leads to packages with names like `utils`.  With something like `foo/bar/tokenutils` you get `tokenutils` as the package name which is far more helpful when used without needing an import alias (`tokenutils.Bar` vs `utils.Bar`).

---

This change looks fine once you squash.  It will likely be next week before approvers have enough bandwidth to look at this since this week is code freeze.

/lgtm

**Comment by boddumanohar** [comment]
@enj Squashed both the commits. Can you please review again?

**Review by dixudx — APPROVED** [review]
/lgtm
/approve

**Comment by boddumanohar** [comment]
Now that the coding period has started, can we merge this PR? 

Thanks,
Manohar.

**Comment by boddumanohar** [comment]
Hi @smarterclayton Can you please take a look at this PR? 

Thanks.

**Comment by boddumanohar** [comment]
Hi @smarterclayton Can you please take a look at this PR?

Thanks.

**Comment by boddumanohar** [comment]
I rebased the PR on latest master. Can you guys please review again? 
cc: @enj @neolit123 

thanks,
Manohar.

**Review by neolit123 — COMMENTED** [review]
/lgtm

the changes under /plugin and /staging still need a top level approver.

**Pull request closed without merging**